text/0005-architecture-reviews.md

```diff
@@ -0,0 +1,163 @@
+  * Open-ended
+  * Made available to the the team prior to the review so they can prepare answers
+
+4. The chair convenes a review meeting. The discussions of this meeting should be carefully minuted. The meeting comprises:
+  * The team making the proposal
+  * The chair
+  * Other senior engineers who will provide valuable input
+
+The review meeting should be a discussion of the issues, but should avoid feeling pressured to make a decision. The goal is to understand the issues raised by the question(s) and the background from the review packet, and to add to it wisdom from the people at the review. The meeting may decide to alter the questions and undergo a subsequent review. We should be very open about conversations that happen after the review meeting.
+
+The ideal review process scales well. The same basic system should work for a quick 2 person decision over the best way to design a certain feature, and for a critical organization-wide decision about a path to take. The term “team” is used above although we strongly recommend design reviews for smaller questions. If you feel yourself wondering if some design is best, it should be easy to perform a review.
+
+For both a Roadmap Review and a Design Review the goal is to hear perspectives that will lead to a better definition of a problem or solution, and better judgment about whether it should be solved or how to solve it.
+
+## Rationale
+
+### Key considerations
+
+* What are the desired **outputs** of the review process?
+* Which **audience** is intended to consume the outputs of the review process?
+* What committee **structure** is best positioned to deliver those outputs?
+* What are the **inputs** that the committee should evaluate?
+* What **process** will most encourage participants to look at review as an asset rather than an obstacle?
+* What kinds of **conduct** are necessary in order for the process to be inclusive and constructive?
+* What size and **maturity** should projects achieve before undergoing review?
+* When should review results be **disseminated**?
+
+### Outputs and Audience
+
+The first two considerations define the goals of the review process and are two sides of the same coin.
+
+The audience of a Roadmap Review is those "up the chain" of decision makers and the audience of a Design Review is those running “down the chain” of implementers.
+
+A Roadmap Review should focus on:
+
+* The problem space
+* Large forces impacting Mozilla’s choices in the space
+* Competitive analysis
+* Opportunity cost
+
+The key outcome is senior management investing in the problem space and building consensus around a shared reality.
+
+A Design Review should focus on:
+
+* The solution space
+* Technical roadmaps
+* Milestones and timelines
+* Resource allocation
+
+The key outcome is middle management and individual engineers understanding the value proposition and the path to success.
```

Comment:
For a design review, I think a key outcome is engineers understanding the design constraints involved and possibly the tradeoffs between them.  This is especially important if implementation reality leads to having to re-evaluate parts of the design.  Having clear documentation for why the design is the way it is helps avoid duplicating a bunch of work in that situation.

Reply:
Yes, I think that makes sense. I put:

> The key outcomes are for middle management to understand the value proposition and the path to success and for engineers to understand the design constraints and the tradeoffs between them (this latter part is important when implementation reality causes a re-think, in which case we want to avoid unnecessary re-work).